error-handling/src/main/java/com/palantir/remoting2/errors/Param.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting2.errors;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/** A wrapper around an argument used to build a formatted message. */
+public abstract class Param<T> {
```

Comment:
Does this need to implement Serializable as we're embedding these in exceptions which inherently are serializable? One example might be an Either returned via Spark.